File: src/com/sterndu/json/DoubleValue.java
package com.sterndu.json;

import java.util.Locale;
import java.util.function.Function;

public class DoubleValue implements NumberValue {

	private double value = 0.0d;

	public DoubleValue(double value) {
		this.value = value;
	}

	@Override
	public Number getValue() { return value; }

	public double getValueDouble() { return value; }

	public void setValue(double value) { this.value = value; }

	@Override
	public String toJson() {
		return String.format(Locale.US, "%g", value);
	}

	@Override
	public String toJson(Function<Object, String> function) {
		return String.format(Locale.US, "%g", value);
	}

	@Override
	public String toString() {
		return toJson();
	}

}

File: src/com/sterndu/json/DoubleValue.kt
@file:JvmName("DoubleValue")
package com.sterndu.json

import java.util.*

class DoubleValue @JvmOverloads constructor(value: Double = 0.0) : NumberValue {
	private var valueDouble = value

	fun getValueDouble(): Double {
		return valueDouble
	}

	override fun getValue(): Number {
		return valueDouble
	}

	fun setValue(value: Double) {
		valueDouble = value
	}

	override fun toJson(): String {
		return String.format(Locale.US, "%g", valueDouble)
	}

	override fun toJson(function: Function1<Any?, String>): String {
		return String.format(Locale.US, "%g", valueDouble)
	}

	override fun toJsonValue(): JsonValue {
		return this
	}

	override fun toJsonValue(function: Function1<Any?, String>): JsonValue {
		return this
	}

	override fun toString(): String {
		return toJson()
	}
}
